exec/java-exec/src/main/java/org/apache/drill/exec/server/rest/WebServer.java

```diff
@@ -495,4 +509,50 @@ private void generateOptionsDescriptionJSFile() throws IOException {
       writer.flush();
     }
   }
+
+  //Generates ACE library javascript populated with list of available SQL functions
+  private void generateFunctionJS() throws IOException {
+    //Naturally ordered set of function names
+    TreeSet<String> functionSet = new TreeSet<>();
+    //Extracting ONLY builtIn functions (i.e those already available)
+    List<FunctionHolder> builtInFuncHolderList = this.drillbit.getContext().getFunctionImplementationRegistry().getLocalFunctionRegistry()
+        .getAllJarsWithFunctionsHolders().get(LocalFunctionRegistry.BUILT_IN);
+
+    //Build List of usable functions
+    int skipCount = 0;
+    for (FunctionHolder builtInFunctionHolder : builtInFuncHolderList) {
+      String name = builtInFunctionHolder.getName();
+      if (!name.contains(" ") && name.matches("([a-z]|[A-Z])\\w+")) {
+        functionSet.add(name);
+      } else {
+        logger.debug("Non-alphabetic leading character. Function skipped : {} ", name);
+        skipCount++;
+      }
+    }
+    logger.debug("{} functions will not be available in WebUI : {} ", skipCount);
+
+    //Generated file
+    File functionsListFile = new File(getTmpJavaScriptDir(), ACE_MODE_SQL_JS);
+    //Template source Javascript file
+    InputStream aceModeSqlTemplateStream = Resource.newClassPathResource(ACE_MODE_SQL_TEMPLATE_JS).getInputStream();
+    functionsListFile.deleteOnExit();
+    int numLeftToWrite = functionSet.size();
+    //Create a copy of a template and write with that!
+    java.nio.file.Files.copy(aceModeSqlTemplateStream, functionsListFile.toPath());
+    StringBuilder funcListBldr = new StringBuilder();
```

Comment:
Much easier logic can be used here: you just append all functions with `|` and in the end of StringBuilder is not empty substract last char.

Reply:
Yeah... this should have been trivial. Silly on my part 🤦‍♂️